common/libraries/util/utils.h

```diff
@@ -27,6 +27,14 @@
 
 /// Convert bit array of size 4 to uint32
 #define BYTE_ARRAY_TO_UINT32(x) ((x)[0] << 24 | (x)[1] << 16 | (x)[2] << 8 | (x)[3])
+/// Read 64-bit value from big-endian serialized byte-array
+#define U64_READ_BE_ARRAY(x)                                                                                   \
+  (((uint64_t)(x)[0] << 56) | ((uint64_t)(x)[1] << 48) | ((uint64_t)(x)[2] << 40) | ((uint64_t)(x)[3] << 32) | \
+   ((x)[4] << 24) | ((x)[5] << 16) | (x)[6] << 8 | (x)[7])
```

Comment:
difficult to grasp. can we leverage `U32_READ_BE_ARRAY` for improved readability :)?